pages/index.vue

```diff
@@ -1,13 +1,92 @@
 <template>
   <div class="flex flex-col h-screen w-screen">
     <GlobalHeader />
-    <div class="flex flex-grow p-4">
-      <div class="flex min-w-[600px]">
+    <div
+      class="flex flex-grow p-4"
+      style="height: 90%;"
+    >
+      <div class="flex min-w-[300px] flex-grow resize-x overflow-auto">
         <CodeEditor />
       </div>
-      <div class="flex flex-grow">
+      <div class="separator" />
+      <div class="flex min-w-[300px] resize-x overflow-auto">
         <GameScreen />
       </div>
     </div>
   </div>
 </template>
+
+<script lang="ts">
+import { defineComponent, ref, onMounted, onUnmounted } from "vue";
+
+export default defineComponent({
+  setup() {
+    const isResizing = ref(false);
+    const initialX = ref(0);
+    const initialWidthEditor = ref(0);
+    const initialWidthGame = ref(0);
+
```

Comment:
Let's use vue ref's to access DOM instead of `.querySelector`.

```suggestion
    const resizeEditorElement = ref<HTMLDivElement | null>(null);
    const resizeGameElement = ref<HTMLDivElement | null>(null);
    const separator = ref<HTMLDivElement | null>(null);
```